nanomq/apps/broker.c

```diff
@@ -443,14 +444,15 @@ server_cb(void *arg)
 							    work, &work->config->bridge);
 						}
 						// dont modify original retain msg;
-						nng_msg *rmsg;
-						nng_msg_alloc(&rmsg, 0);
+						nng_msg_clone(rmsg);
+						nng_msg_set_proto_data(rmsg, NULL, proto_data);
```

Comment:
It's a trick! Some comments should be added to avoid using it casually

Reply:
yup